src/vis/heatmap/HeatmapVisSidebar.tsx

```diff
@@ -17,7 +17,11 @@ export function HeatmapVisSidebar({
   return (
     <>
       <MultiSelect
-        callback={(catColumnsSelected: ColumnInfo[]) => setConfig({ ...config, catColumnsSelected })}
+        callback={(catColumnsSelected: ColumnInfo[]) => {
+          // For the heatmap, only 2 columns can be selected at once
+          const catColumns = catColumnsSelected.length > 2 ? catColumnsSelected.slice(catColumnsSelected.length - 2) : catColumnsSelected;
```

Comment:
Currently, you are using always the last two category columns, which could be written like this:


```suggestion
          // For the heatmap, only 2 columns can be selected at once
          const catColumns = catColumnsSelected.length > 2 ? catColumnsSelected.slice(-2) : catColumnsSelected;
```

If you would like to use the first two categories, it must be `catColumnsSelected.slice(2)` instead.

Reply:
we'll do that in a follow-up pr